app/assets/stylesheets/views/_homepage.scss

```diff
@@ -216,108 +216,134 @@ body.homepage {
       }
     }
     form#header-search {
-      padding-bottom: 30px;
+      @extend %contain-floats;
+      margin-bottom: 30px;
 
-      .header-search-content {
-        @extend %contain-floats;
-        position: relative;
-        background: #fff;
-      }
       label {
-        float: left;
-        height: 40px;
-        line-height: 40px;
-        text-indent: 15px;
-        overflow: hidden;
-        display: block;
-        margin-right: 10px;
-        color: $secondary-text-colour;
+        color: $white;
+      }
 
-        .js-enabled & {
-          float: none;
-          position: absolute;
-          left: 0;
-          top: 1px;
-          z-index: 1;
+      .search-input-wrapper {
+        float: left;
+        width: 100%;
+        
+        .search-input {
           width: 100%;
-          margin-right: 0;
+          margin: 0.5em 0;
+          padding: 6px;
+
+          @include core-19($line-height: (28 / 19), $line-height-640: (28 / 16));
         }
       }
-      input {
-        @include box-sizing(border-box);
+
+      .submit {
         float: left;
-        width: 50%;
-        min-width: 0;
-        display: block;
-        margin: 0;
+        padding: 6px;
+        background: $black;
+        color: $white;
         border: 0;
-        height: 40px;
-        padding: 6px 0;
-        @include ie-lte(7){
-          height: 28px;
-        }
-        @include appearance(none);
       }
 
-      input#search-main {
-        @include core-19($line-height: (28/19), $line-height-640: (28/16));
-        position: relative;
-        padding: 6px;
-        z-index: 3;
-        background: transparent;
+      // IE6 + IE7 always get the simpler version, regardless of whether
+      // javascript is enabled
+      @if ($is-ie == false) or ($ie-version >= 8) {
+        .search-input-wrapper {
+          @include box-sizing(border-box);
 
-        &.focus,
-        &:focus {
-          background: #fff;
-        }
+          padding-right: 40px;
+          margin-right: -40px;
```

Comment:
Minor: To make it clearer where the `40px` are coming from, you could extract that into a (local) sass variable (e.g. `$search-icon-size`). That is mostly to improve readability and therefore maintainability of the code.